flax/linen/module.py

```diff
@@ -676,18 +749,41 @@ def _verify_single_or_no_compact(cls):
       raise errors.MultipleMethodsCompactError()
 
   @classmethod
-  def _wrap_module_methods(cls):
-    """Wraps user-defined non-inherited methods with state management functions."""
+  def _wrap_module_attributes(cls):
+    """Wraps user-defined non-inherited methods and descriptors with state
+    management functions.
+    """
     exclusions = ([f.name for f in dataclasses.fields(cls)] +
                   ['__eq__', '__repr__', '__init__', '__hash__',
-                   '__post_init__'])
+                   '__post_init__', '__dict__'])
     for key in _get_local_method_names(cls, exclude=exclusions):
       method = getattr(cls, key)
       if hasattr(method, 'nowrap'):
         continue
       setattr(cls, key, wrap_method_once(method))
+    for key in _get_local_descriptor_names(cls, exclusions):
```

Comment:
why use the same exclusions as for methods? the callability filter will rules all those out, no?